internal/keppel/rbac_policy.go

```diff
@@ -20,16 +20,113 @@
 package keppel
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+
 	"github.com/sapcc/go-bits/regexpext"
 )
 
-// NewRBACPolicy is a policy granting user-defined access to repos in an account.
+// RBACPolicy is a policy granting user-defined access to repos in an account.
 // It is stored in serialized form in the RBACPoliciesJSON field of type Account.
-//
-// TODO: rename to type RBACPolicy after the `rbac_policies` table has been removed
-type NewRBACPolicy struct {
-	CidrPattern       string                `json:"match_cidr,omitempty"`
-	RepositoryPattern regexpext.PlainRegexp `json:"match_repository,omitempty"`
-	UserNamePattern   regexpext.PlainRegexp `json:"match_username,omitempty"`
-	Permissions       []string              `json:"permissions"`
+type RBACPolicy struct {
+	CidrPattern       string                  `json:"match_cidr,omitempty"`
+	RepositoryPattern regexpext.BoundedRegexp `json:"match_repository,omitempty"`
+	UserNamePattern   regexpext.BoundedRegexp `json:"match_username,omitempty"`
+	Permissions       []RBACPermission        `json:"permissions"`
+}
+
+// RBACPermission enumerates permissions that can be granted by an RBAC policy.
+type RBACPermission string
+
+const (
+	PolicyAllowsPull               RBACPermission = "pull"
+	PolicyAllowsPush               RBACPermission = "push"
+	PolicyAllowsDelete             RBACPermission = "delete"
+	PolicyAllowsAnonymousPull      RBACPermission = "anonymous_pull"
+	PolicyAllowsAnonymousFirstPull RBACPermission = "anonymous_first_pull"
+)
```

Comment:
FYI: I'm not entirely happy with the names for these enum variants, but it's the best I could come up with. Anything that contains "RBAC" ends up looking like a keysmash, e.g. `AnonymousFirstPullRBACPermission`.

Reply:
Maybe we can drop the Allows and just take that for implicit